<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>962</width>
    <height>750</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>CGAL Periodic Delaunay Triangulation</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="Viewer" name="viewer" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>962</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionLoad_Points"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>&amp;Options</string>
    </property>
    <addaction name="actionWireframe"/>
    <addaction name="actionPlanar_triangulation"/>
    <addaction name="separator"/>
    <addaction name="actionDraw_1_sheeted_covering"/>
    <addaction name="actionDraw_bordering_cells_multiply"/>
    <addaction name="separator"/>
    <addaction name="actionDraw_segments"/>
    <addaction name="actionDraw_triangles"/>
    <addaction name="actionDraw_tetrahedra"/>
    <addaction name="separator"/>
    <addaction name="actionDraw_cube_square"/>
    <addaction name="separator"/>
    <addaction name="actionClip_along_the_cube_square"/>
    <addaction name="action2_color_clipping"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuInitialization">
    <property name="title">
     <string>&amp;Initialization</string>
    </property>
    <addaction name="actionEmpty_scene"/>
    <addaction name="actionSingle_Point"/>
    <addaction name="actionRandom_Point_Set"/>
    <addaction name="actionRandom_Points_in_Plane"/>
    <addaction name="actionPoint_grid"/>
   </widget>
   <widget class="QMenu" name="menuFeatures">
    <property name="title">
     <string>&amp;Features</string>
    </property>
    <addaction name="actionPoint_location"/>
    <addaction name="actionConflict_region"/>
   </widget>
   <widget class="QMenu" name="menuActions">
    <property name="title">
     <string>&amp;Actions</string>
    </property>
    <addaction name="actionFlying_ball"/>
    <addaction name="actionPause"/>
    <addaction name="separator"/>
    <addaction name="actionInsert_point"/>
    <addaction name="actionInsert_random_point"/>
    <addaction name="separator"/>
    <addaction name="actionGrab_image"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionDemo_Help"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
    <addaction name="actionAbout_CGAL"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuInitialization"/>
   <addaction name="menuActions"/>
   <addaction name="menuFeatures"/>
   <addaction name="menuOptions"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLoad_Points"/>
   <addaction name="actionGrab_image"/>
   <addaction name="separator"/>
   <addaction name="actionEmpty_scene"/>
   <addaction name="actionSingle_Point"/>
   <addaction name="actionRandom_Point_Set"/>
   <addaction name="actionRandom_Points_in_Plane"/>
   <addaction name="actionPoint_grid"/>
   <addaction name="separator"/>
   <addaction name="actionPause"/>
   <addaction name="actionFlying_ball"/>
   <addaction name="separator"/>
   <addaction name="actionInsert_point"/>
   <addaction name="actionInsert_random_point"/>
   <addaction name="separator"/>
   <addaction name="actionPoint_location"/>
   <addaction name="actionConflict_region"/>
   <addaction name="separator"/>
   <addaction name="actionWireframe"/>
   <addaction name="actionPlanar_triangulation"/>
   <addaction name="separator"/>
   <addaction name="actionDraw_1_sheeted_covering"/>
   <addaction name="actionDraw_bordering_cells_multiply"/>
   <addaction name="actionClip_along_the_cube_square"/>
   <addaction name="action2_color_clipping"/>
  </widget>
  <action name="actionLoad_Points">
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/fileOpen.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/fileOpen.png</iconset>
   </property>
   <property name="text">
    <string>Load Points</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionWireframe">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/wireframe.png</normaloff>
     <normalon>:/cgal/Periodic_3_triangulation_3/icons/wireframeOff.png</normalon>:/cgal/Periodic_3_triangulation_3/icons/wireframe.png</iconset>
   </property>
   <property name="text">
    <string>Wireframe</string>
   </property>
   <property name="toolTip">
    <string>Switches wireframe rendering on and off</string>
   </property>
   <property name="shortcut">
    <string>X</string>
   </property>
  </action>
  <action name="actionPlanar_triangulation">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/planar.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/planar.png</iconset>
   </property>
   <property name="text">
    <string>Planar triangulation</string>
   </property>
   <property name="toolTip">
    <string>Shows only simplices for which z=0 everywhere</string>
   </property>
   <property name="shortcut">
    <string>Z</string>
   </property>
  </action>
  <action name="actionDraw_1_sheeted_covering">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/cover1.png</normaloff>
     <normalon>:/cgal/Periodic_3_triangulation_3/icons/cover27.png</normalon>:/cgal/Periodic_3_triangulation_3/icons/cover1.png</iconset>
   </property>
   <property name="text">
    <string>Draw 1-sheeted covering</string>
   </property>
   <property name="toolTip">
    <string>toggles between displaying only one periodic copy of each simplex and the internal representation </string>
   </property>
   <property name="shortcut">
    <string>1</string>
   </property>
  </action>
  <action name="actionDraw_bordering_cells_multiply">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/multiple.png</normaloff>
     <normalon>:/cgal/Periodic_3_triangulation_3/icons/multipleOff.png</normalon>:/cgal/Periodic_3_triangulation_3/icons/multiple.png</iconset>
   </property>
   <property name="text">
    <string>Draw bordering cells multiply</string>
   </property>
   <property name="toolTip">
    <string>Draws each simplex for which the intersection with the fundamental cube is non-empty</string>
   </property>
   <property name="shortcut">
    <string>M</string>
   </property>
  </action>
  <action name="actionClip_along_the_cube_square">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/clipping.png</normaloff>
     <normalon>:/cgal/Periodic_3_triangulation_3/icons/clippingOff.png</normalon>:/cgal/Periodic_3_triangulation_3/icons/clipping.png</iconset>
   </property>
   <property name="text">
    <string>Clip along the cube/square</string>
   </property>
  </action>
  <action name="action2_color_clipping">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/twoColorClipping.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/twoColorClipping.png</iconset>
   </property>
   <property name="text">
    <string>2-color clipping</string>
   </property>
   <property name="toolTip">
    <string>show the clipped part of all clipped edges in a different color</string>
   </property>
  </action>
  <action name="actionPoint_grid">
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/initGrid.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/initGrid.png</iconset>
   </property>
   <property name="text">
    <string>Point grid</string>
   </property>
   <property name="iconText">
    <string>pg</string>
   </property>
   <property name="toolTip">
    <string>A grid of 36 points that can be triangulated in 1-sheeted covering space</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionSingle_Point">
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/init1.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/init1.png</iconset>
   </property>
   <property name="text">
    <string>Single point</string>
   </property>
   <property name="shortcut">
    <string>F2</string>
   </property>
  </action>
  <action name="actionRandom_Points_in_Plane">
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/initRandPlanar.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/initRandPlanar.png</iconset>
   </property>
   <property name="text">
    <string>Random planar point set</string>
   </property>
   <property name="toolTip">
    <string>Initial triangulation of 10 points chosen at random with z=0</string>
   </property>
   <property name="shortcut">
    <string>F4</string>
   </property>
  </action>
  <action name="actionRandom_Point_Set">
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/initRand.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/initRand.png</iconset>
   </property>
   <property name="text">
    <string>Random point set</string>
   </property>
   <property name="toolTip">
    <string>Initial triangulation of 30 points chosen at random</string>
   </property>
   <property name="shortcut">
    <string>F3</string>
   </property>
  </action>
  <action name="actionEmpty_scene">
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/init0.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/init0.png</iconset>
   </property>
   <property name="text">
    <string>Empty scene</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionDraw_cube_square">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Draw domain (cube/square)</string>
   </property>
   <property name="shortcut">
    <string>D</string>
   </property>
  </action>
  <action name="actionPoint_location">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/locate.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/locate.png</iconset>
   </property>
   <property name="text">
    <string>Point location</string>
   </property>
   <property name="toolTip">
    <string>Marks the tetrahedron that currently contains the flying ball</string>
   </property>
   <property name="shortcut">
    <string>L</string>
   </property>
  </action>
  <action name="actionConflict_region">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/conflict.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/conflict.png</iconset>
   </property>
   <property name="text">
    <string>Conflict region</string>
   </property>
   <property name="toolTip">
    <string>Shows all tetrahedra in whose circumcircle the flying ball is contained</string>
   </property>
   <property name="shortcut">
    <string>C</string>
   </property>
  </action>
  <action name="actionHole">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Hole*</string>
   </property>
   <property name="shortcut">
    <string>H</string>
   </property>
  </action>
  <action name="actionStar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Star*</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
  </action>
  <action name="actionFlying_ball">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/ball.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/ball.png</iconset>
   </property>
   <property name="text">
    <string>Flying ball</string>
   </property>
   <property name="toolTip">
    <string>Shows a ball flying through the cube, showing the periodicity</string>
   </property>
   <property name="shortcut">
    <string>B</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/pause.png</normaloff>
     <normalon>:/cgal/Periodic_3_triangulation_3/icons/play.png</normalon>:/cgal/Periodic_3_triangulation_3/icons/pause.png</iconset>
   </property>
   <property name="text">
    <string>Pause</string>
   </property>
   <property name="toolTip">
    <string>Pauses the flying ball</string>
   </property>
   <property name="shortcut">
    <string>P</string>
   </property>
  </action>
  <action name="actionInsert_point">
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/inputPointBall.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/inputPointBall.png</iconset>
   </property>
   <property name="text">
    <string>Insert point</string>
   </property>
   <property name="toolTip">
    <string>Insert point at the position of the flying ball (whether it is shown or not)</string>
   </property>
   <property name="shortcut">
    <string>I</string>
   </property>
  </action>
  <action name="actionInsert_random_point">
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/inputPointRandom.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/inputPointRandom.png</iconset>
   </property>
   <property name="text">
    <string>Insert random point</string>
   </property>
   <property name="toolTip">
    <string>Insert one point at random position (within the cube)</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
  </action>
  <action name="actionDraw_segments">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Draw segments</string>
   </property>
   <property name="shortcut">
    <string>2</string>
   </property>
  </action>
  <action name="actionDraw_triangles">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Draw triangles</string>
   </property>
   <property name="shortcut">
    <string>3</string>
   </property>
  </action>
  <action name="actionDraw_tetrahedra">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Draw tetrahedra</string>
   </property>
   <property name="shortcut">
    <string>4</string>
   </property>
  </action>
  <action name="actionGrab_image">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="Periodic_3_triangulation_3.qrc">
     <normaloff>:/cgal/Periodic_3_triangulation_3/icons/camera.png</normaloff>:/cgal/Periodic_3_triangulation_3/icons/camera.png</iconset>
   </property>
   <property name="text">
    <string>Grab image</string>
   </property>
   <property name="shortcut">
    <string>G</string>
   </property>
  </action>
  <action name="actionExport_pov">
   <property name="text">
    <string>Export pov</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionDemo_Help">
   <property name="text">
    <string>Demo Manual</string>
   </property>
   <property name="shortcut">
    <string>H</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="shortcut">
    <string>A</string>
   </property>
  </action>
  <action name="actionAbout_CGAL">
   <property name="text">
    <string>About CGAL</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Viewer</class>
   <extends>QWidget</extends>
   <header>Viewer.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="Periodic_3_triangulation_3.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionClip_along_the_cube_square</sender>
   <signal>toggled(bool)</signal>
   <receiver>action2_color_clipping</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
